io-sim/test/Main.hs

```diff
@@ -12,5 +12,5 @@ tests :: TestTree
 tests =
   testGroup "IO Sim"
   [ Test.IOSim.tests
-  , Test.Control.Monad.Class.MonadMVar.tests
+  -- , Test.Control.Monad.Class.MonadMVar.tests
```

Comment:
Whoops, I think we turned these off because they were flaky but not for any reason that we could see related to anything we did. @dcoutts any clue?